src/number_theory/bernoulli.lean

```diff
@@ -258,3 +258,112 @@ begin
   { ext x, rw bernoulli_eq_bernoulli' (succ_ne_zero x ∘ succ.inj) },
   { ring },
 end
+
+theorem bernoulli_power_series :
+  (exp ℚ - 1) * mk (λ n, (bernoulli n / n!)) = X :=
+begin
+  have h: power_series.mk (λ n, (bernoulli' n / n! : ℚ)) * (exp ℚ - 1)
+    = X * exp ℚ :=
+  begin
+    simp only [bernoulli'_power_series],
+  end,
+  sorry,
+end
+
+section faulhaber
+
+lemma aux_cauchy_prod (n : ℕ) :
+mk (λ (p : ℕ), bernoulli p / ↑p!) *
+ mk (λ (q : ℕ), (coeff ℚ (q + 1)) (exp ℚ ^ n))
+ = mk (λ (p : ℕ), (finset.range (p + 1)).sum (λ (i : ℕ),
+ bernoulli i * ↑((p + 1).choose i) * ↑n ^ (p + 1 - i) / ↑(p + 1)!)) :=
+begin
```

Comment:
May I suggest restructuring this proof as follows? I'll include an explanation of the changes I made below. Similar changes can also be made to `aux_power_series_equal` and `faulhaber`, but I've held off on saying anything in case you want to try it yourself.
```
begin
  let f := λ a, λ b, bernoulli a / ↑a! * (coeff ℚ (b + 1)) (exp ℚ ^ n),
  ext q,
  simp only [power_series.coeff_mul, coeff_mk, cast_mul, nat.sum_antidiagonal_eq_sum_range_succ f],
  refine sum_congr rfl _,
  simp_intros m h only [finset.mem_range],
  have hq : (q.succ! : ℚ) ≠ 0 := by exact_mod_cast q.succ.factorial_ne_zero,
  have hqm1 : q - m + 1 = q + 1 - m,
  { rw nat.sub_add_comm,
    exact le_of_lt_succ h },
  simp only [f, exp_pow_eq_rescale_exp, rescale, one_div, coeff_mk, ring_hom.coe_mk, coeff_exp,
    ring_hom.id_apply, cast_mul, rat.algebra_map_rat_rat],
  rw [choose_eq_factorial_div_factorial h.le, eq_comm, div_eq_iff hq, succ_eq_add_one,
    mul_assoc _ _ ↑(q.succ!), mul_comm _ ↑(q.succ!), ←mul_assoc, div_mul_eq_mul_div,
    mul_comm (n ^ (q - m + 1) : ℚ), ←mul_assoc _ _ (n ^ (q - m + 1) : ℚ), ←one_div, mul_one_div,
    div_div_eq_div_mul, hqm1, cast_dvd, cast_mul],
  { ring },
  { exact factorial_mul_factorial_dvd_factorial h.le },
  { simp [not_false_iff, factorial_ne_zero] },
end
```
- I reorder the proof, moving the `have` statements earlier. This allows me to consolidate the numerous `simp`s and `rw`s into one statement each.
- You have `finest` and `nat` open. That means you can remove most of the `finest.` and `nat.` "prefixes".
- `le_of_lt h` becomes `h.le`
- `norm_cast, exact q.succ.factorial_ne_zero,` becomes `exact_mod_cast q.succ.factorial_ne_zero,`
- I use `simp_intros m h` instead of `intros m h, simp at h`
- I also make a few minor style corrections

Reply:
I tried to make all suggested changes everywhere..